### PR TITLE
Guide for UaaS JWT authorisation 

### PR DESCRIPTION
**Pull request by laurencemcknight**
image

### CODE REVIEW

File: src/content/guides/integrating-third-party-asset.mdx [hunk 1]
```diff
@@ -108,6 +108,29 @@ curl https://auth.centrapay.com/api/.well-known/jwks.json
 }
 ```
 
+To verify a request start by decripting the JWT.
```
Comment (tom-blake) [inline src/content/guides/integrating-third-party-asset.mdx:111]: @laurencemcknight Typo here - "decripting"

File: src/content/guides/integrating-third-party-asset.mdx [hunk 2]
```diff
@@ -108,6 +108,29 @@ curl https://auth.centrapay.com/api/.well-known/jwks.json
...
+
+**Decripted JWT Payload**
```
Comment (tom-blake) [inline src/content/guides/integrating-third-party-asset.mdx:113]: @laurencemcknight There is a sentence above about verifying the JWT using the public key. I think this needs restructuring - worth looking at how other companies document this (e.g. [Stripe](https://stripe.com/docs/webhooks/signatures))

File: src/content/guides/integrating-third-party-asset.mdx [hunk 3]
```diff
@@ -108,6 +108,29 @@ curl https://auth.centrapay.com/api/.well-known/jwks.json
...
+  "aud": "https://your.endpoint/",
+  "request_body_sha256": "b9195bf41bf0e38ab0ab44e7ef5b9af5cb0fe2ece8dee5d112d7485bf4ef0007",
+}
+```
+
+Use the decripted JWT validate the following fields:
+
+|        Field        |                                                                         Assertion                                                                           |
+| ------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| exp                 | Assert that exp is greater than the current Unix timestamp.                                                                                                 |
```
Comment (tom-blake) [inline src/content/guides/integrating-third-party-asset.mdx:128]: @laurencemcknight I thought we were including the `iat` field too?

We should document what our expiry is (e.g. 5 mins from the time that the JWT was created) + provide the `iat` field to allow integrators to use a shorter expiry if they wish

File: src/content/guides/integrating-third-party-asset.mdx [hunk 4]
```diff
@@ -108,6 +108,29 @@ curl https://auth.centrapay.com/api/.well-known/jwks.json
...
+| aud                 | Assert that aud is equal to the base URL of your Uplink API.                                                                                                |
+| request_body_sha256 | Hash the request payload using the [SHA256 algorythm](https://en.wikipedia.org/wiki/SHA-2). Assert that request_body_sha256 is equal to the hashed payload. |
```
Comment (tom-blake) [inline src/content/guides/integrating-third-party-asset.mdx:130]: @laurencemcknight We should be really explicit with should integrators must do - verifying the payload is important

File: src/content/guides/integrating-third-party-asset.mdx (hunk 1 above)
Comment (IsaacMurtagh) [reply src/content/guides/integrating-third-party-asset.mdx:111]: Decrypting is not the right word. Decoding is more accurate

File: src/content/guides/integrating-third-party-asset.mdx (hunk 4 above)
Comment (IsaacMurtagh) [reply src/content/guides/integrating-third-party-asset.mdx:130]: @laurencemcknight You could give an example how to do it in JS as it's important they know they need to stringify the json

File: src/content/guides/integrating-third-party-asset.mdx [hunk 5]
```diff
@@ -108,6 +108,29 @@ curl https://auth.centrapay.com/api/.well-known/jwks.json
...
 
+To verify a request start by decripting the JWT.
+
+**Decripted JWT Payload**
+
+```JSON
+{
+  "iat": "1684105185",
+  "exp": "1684105485",
+  "aud": "https://your.endpoint/",
```
Comment (IsaacMurtagh) [inline src/content/guides/integrating-third-party-asset.mdx:119]: Can you confirm we will be sending a trailing slash?

File: src/content/guides/integrating-third-party-asset.mdx [hunk 6]
```diff
@@ -108,6 +108,29 @@ curl https://auth.centrapay.com/api/.well-known/jwks.json
...
+  "request_body_sha256": "b9195bf41bf0e38ab0ab44e7ef5b9af5cb0fe2ece8dee5d112d7485bf4ef0007",
+}
+```
+
+Use the decripted JWT validate the following fields:
+
+|        Field        |                                                                         Assertion                                                                           |
```
Comment (IsaacMurtagh) [inline src/content/guides/integrating-third-party-asset.mdx:126]: I think this table is better used for explaining the value and then a paragraph on each of the fields about how to make the assertion.

**Review by IsaacMurtagh — APPROVED** [review]

File: src/content/guides/integrating-third-party-asset.mdx (hunk 1 above)
Comment (daallen303) [reply src/content/guides/integrating-third-party-asset.mdx]: Hmm, I think it's more decrypting. We encrypt the message with a specific private key and then they need the public key in order to decrypt the message right?

Comment (daallen303) [reply src/content/guides/integrating-third-party-asset.mdx]: I think encoding/decoding seems more like applying an algorithm without any private/public keys.

Comment (laurencemcknight) [reply src/content/guides/integrating-third-party-asset.mdx]: @daallen303 "Encoding" and "decoding" is the common terminology online used for JWTs. Additionally we already use that terminology in our docs for JWTs.

**Pull request merged**